reVX/turbine_flicker/turbine_flicker.py

```diff
@@ -0,0 +1,563 @@
+        """
+        Exclude all pixels that will cause flicker exceeding the
+        "flicker_threshold" on any building in "building_layer". Buildings
+        are defined as pixels with >= the "building_threshold value in
+        "building_layer". Shadow flicker is computed at the supply curve point
+        resolution and applied to all buildings within that supply curve point
+        sub-array.
+
+        Parameters
+        ----------
+        gid : int
+            Supply curve point gid to aggregate wind directions for
+        excl_fpath : str
+            Filepath to exclusions h5 file. File must contain "tm_dset".
+        res_fpath : str
+            Filepath to wind resource .h5 file containing hourly wind
+            direction data
+        building_layer : str
+            Exclusion layer containing buildings from which turbine flicker
+            exclusions will be computed.
+        hub_height : int
+            Hub-height in meters to compute turbine shadow flicker for
+        building_threshold : float, optional
+            Threshold for exclusion layer values to identify pixels with
+            buildings, values are % of pixel containing a building,
+            by default 0
+        flicker_threshold : int, optional
+            Maximum number of allowable flicker hours, by default 30
+        tm_dset : str, optional
+            Dataset / layer name for wind toolkit techmap,
+            by default 'techmap_wtk'
+        resolution : int, optional
+            SC resolution, must be input in combination with gid,
+            by default 128
+
+        Returns
+        -------
+        excl_row_idx : ndarray
+            Axis 0 indices of pixels to be excluded
+        excl_col_idx : ndarray
+            Axis 1 indices of pixels to be excluded
+        """
+        row_idx, col_idx, shape = cls._get_building_indices(
+            excl_fpath, building_layer, gid,
+            resolution=resolution, building_threshold=building_threshold)
+
+        meta, wind_dir = cls._aggregate_wind_dirs(gid,
+                                                  excl_fpath,
+                                                  res_fpath,
+                                                  hub_height,
+                                                  tm_dset=tm_dset,
+                                                  resolution=resolution,
+                                                  exclusion_shape=shape)
+        blade_length = hub_height / 2.5
+        shadow_flicker = cls._compute_shadow_flicker(meta['latitude'],
+                                                     meta['longitude'],
+                                                     blade_length,
+                                                     wind_dir)
+
+        row_shifts, col_shifts = cls._threshold_flicker(
+            shadow_flicker, flicker_threshold=flicker_threshold)
+
+        excl_row_idx = (row_idx + row_shifts[:, None]).ravel()
```

Comment:
Once again this is really hard to think about without just seeing but if you've checked the outputs visually this is great!!